scripts/gen_app_partitions.py

```diff
@@ -237,6 +260,8 @@ def main():
 
     if args.directory is not None:
         parse_obj_files(partitions)
```

Comment:
Do we need to retain the parse_obj_files function?

Reply:
Not for Zephyr use-case afaik, but i'm unsure if anyone downstream is using this script also, so I didn't wanted to remove it straight away.